diff -Nuar linux-6.11.11.orig/scripts/Makefile.lib linux-6.11.11/scripts/Makefile.lib
--- linux-6.11.11.orig/scripts/Makefile.lib	2024-12-05 07:54:34.000000000 -0500
+++ linux-6.11.11/scripts/Makefile.lib	2024-12-08 20:10:26.144185468 -0500
@@ -259,6 +259,7 @@
 
 dtc_cpp_flags  = -Wp,-MMD,$(depfile).pre.tmp -nostdinc                    \
 		 $(addprefix -I,$(DTC_INCLUDE))                          \
+		 $(addprefix -I,$(DTC_INCLUDE)/arm/allwinner)            \
 		 -undef -D__DTS__
 
 ifdef CONFIG_OBJTOOL
