diff -Nuar linux-6.11.11.orig/drivers/input/keyboard/matrix_keypad.c linux-6.11.11/drivers/input/keyboard/matrix_keypad.c
--- linux-6.11.11.orig/drivers/input/keyboard/matrix_keypad.c	2024-12-05 07:54:34.000000000 -0500
+++ linux-6.11.11/drivers/input/keyboard/matrix_keypad.c	2024-12-08 20:12:10.240783158 -0500
@@ -118,8 +118,14 @@
 
 	memset(new_state, 0, sizeof(new_state));
 
-	for (row = 0; row < pdata->num_row_gpios; row++)
-		gpio_direction_input(pdata->row_gpios[row]);
+	// fv - revert kernel commit 01c84b03d80aab9f04c4e3e1f9085f4202ff7c29
+	//      Author: Siarhei Volkau <lis8215@gmail.com>
+	//      Date:   Thu Nov 3 13:40:02 2022 -0700
+	//
+	//          Input: matrix_keypad - force switch rows to input mode
+	//
+	//for (row = 0; row < pdata->num_row_gpios; row++)
+	//	gpio_direction_input(pdata->row_gpios[row]);
 
 	/* assert each column and read the row status out */
 	for (col = 0; col < pdata->num_col_gpios; col++) {
