diff -Naur linux-6.12.19.orig/drivers/input/keyboard/matrix_keypad.c linux-6.12.19/drivers/input/keyboard/matrix_keypad.c
--- linux-6.12.19.orig/drivers/input/keyboard/matrix_keypad.c	2025-03-13 08:02:20.000000000 -0400
+++ linux-6.12.19/drivers/input/keyboard/matrix_keypad.c	2025-03-19 22:24:32.178350585 -0400
@@ -118,8 +118,14 @@
 
 	memset(new_state, 0, sizeof(new_state));
 
-	for (row = 0; row < keypad->num_row_gpios; row++)
-		gpiod_direction_input(keypad->row_gpios[row]);
+	// fv - revert kernel commit 01c84b03d80aab9f04c4e3e1f9085f4202ff7c29
+	//      Author: Siarhei Volkau <lis8215@gmail.com>
+	//      Date:   Thu Nov 3 13:40:02 2022 -0700
+	//
+	//          Input: matrix_keypad - force switch rows to input mode
+	//
+	//for (row = 0; row < keypad->num_row_gpios; row++)
+	//	gpiod_direction_input(keypad->row_gpios[row]);
 
 	/* assert each column and read the row status out */
 	for (col = 0; col < keypad->num_col_gpios; col++) {
