diff -Naur setserial.orig/configure setserial/configure
--- setserial.orig/configure	2000-01-27 10:03:36.000000000 -0500
+++ setserial/configure	2024-11-03 14:10:28.639031039 -0500
@@ -689,7 +689,7 @@
 #line 690 "configure"
 #include "confdefs.h"
 
-main(){return(0);}
+int main(){return(0);}
 EOF
 if { (eval echo configure:695: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   ac_cv_prog_cc_works=yes
