diff -Naur python-bottle-sqlite-0.2.0.orig/bottle_sqlite.py python-bottle-sqlite-0.2.0/bottle_sqlite.py
--- python-bottle-sqlite-0.2.0.orig/bottle_sqlite.py	2020-10-03 11:19:37.000000000 -0400
+++ python-bottle-sqlite-0.2.0/bottle_sqlite.py	2024-12-06 20:12:29.168309351 -0500
@@ -111,8 +111,8 @@
 
         # Test if the original callback accepts a 'db' keyword.
         # Ignore it if it does not need a database handle.
-        argspec = inspect.getargspec(_callback)
-        if keyword not in argspec.args:
+        signature = inspect.signature(_callback)
+        if keyword not in signature.parameters:
             return callback
 
         def wrapper(*args, **kwargs):
